### ISSUE TITLE
`reconcile_compare` incorrectly updating jobs

### ISSUE THREAD

**Issue by OJWatson** [mrc-ide/didehpc#99]
Hey, 

`reconcile_compare` is marking jobs as ERROR due to `tasks_ids` not being found within `dat`:

Sorry it's not the best reprex, but debugging through the problem appears as the names in `dat$name` have extra text outside of the actual task_id. For example, the following task_id (`0fb7386995fe5c6928509e6ada020386`) was marked as ERROR after I ran `reconcile` but I could see the task was still running as it was producing output. So debugging through I could see that the task_id was in `dat$name` but it was enclosed within `TWN(...)` and is the 2nd task in `dat` below. Hopefully, below shows what is going on clearer:

```
Browse[2]> task_id <- "0fb7386995fe5c6928509e6ada020386"
Browse[2]> st_ctx <- obj$task_status(task_id)
Browse[2]> st_ctx
0fb7386995fe5c6928509e6ada020386 
                         "ERROR" 
Browse[2]> 
debug: if (is.null(task_ids)) {
    task_ids <- obj$task_list()
}
Browse[2]> dat <- obj$client$status_user("*", obj$config$cluster)
Browse[2]> head(dat)
   dide_id                                   name   status resources  user          time_start         time_submit            time_end template
1 16894807 MAC (c48381c17232bd3dae07cd323f14dd90) COMPLETE    1 core ow813 2021-08-22 15:34:19 2021-08-22 15:34:19 2021-08-22 15:37:00   24Core
2 16894806 TWN (0fb7386995fe5c6928509e6ada020386)  RUNNING    1 core ow813 2021-08-22 15:34:19 2021-08-22 15:34:19                <NA>   24Core
3 16894805 HKG (7622c928219c9ec567d429c865c81c35)  RUNNING    1 core ow813 2021-08-22 15:34:18 2021-08-22 15:34:18                <NA>   24Core
4 16894804 GUM (39c1399dfcfbd4bd2fd797010127e2fc)    ERROR    1 core ow813 2021-08-22 15:34:18 2021-08-22 15:34:18 2021-08-22 15:43:11   24Core
5 16894803 CUW (6712dcbefb9b263ebde6f94df19c5f21)    ERROR    1 core ow813 2021-08-22 15:34:17 2021-08-22 15:34:17 2021-08-22 15:43:11   24Core
6 16894802 ABW (f5abd5c4a584ea751d007d8fce0e03e7)    ERROR    1 core ow813 2021-08-22 15:34:17 2021-08-22 15:34:17 2021-08-22 15:44:00   24Core
Browse[2]> 
debug: st_ctx <- obj$task_status(task_ids)
Browse[2]> match(task_id, dat$name)
[1] NA
Browse[2]> grep(task_id, dat$name)
[1] 2
``` 

To possibly help explain why the `dat$name` are formatted weirdly, these jobs were submitted as follows, with the `tasks` object being a named character vector of parameters to be passed to `orderly::orderly_bundle`:

```
grp <- obj$lapply(tasks, orderly::orderly_bundle_run, workdir = workdir)
head(tasks)
tasks[1:2]
                                                                             AFG 
"L:\\OJ\\glodide\\analysis\\data\\raw\\2021-08-19\\20210822-153527-3c62797f.zip" 
                                                                             AGO 
"L:\\OJ\\glodide\\analysis\\data\\raw\\2021-08-19\\20210822-153532-56d8758c.zip" 
```

So the named vector is causing an issue here that then causes reconcile to not find them. So don't whether it makes sense to change the `match` call in `reconcile_compare` to be something that uses grep to find the task_id within the names, or perhaps change this at the point where `dat$name` is being created to strip the name before creating the task_id.

**Comment by richfitz** [comment]
Thanks OJ, will take a look at this soon

**Comment by richfitz** [comment]
Fixed in #100
